### PR TITLE
Fix code smells reported by SonarCloud

### PR DESCRIPTION
**Pull request by PeterSuna**
- Add parametrized type to generic
 - Merge this if statement with the enclosing one
 - Define a constant instead of duplicating this litera
 - Remove deprecated class DeclaredEffectiveStatementBase
   Used AbstractDeclaredEffectiveStatement and its subclasses instead
 - Fix Invoke method(s) only conditionally in LOG.info
 - Remove duplicate code
 - Fix cognitive complexity in all methods
 
 SonarCloud after fixed reported issues:
 https://sonarcloud.io/project/issues?id=PeterSuna_lighty-yang-validator&resolved=false&severities=CRITICAL%2CMAJOR&types=CODE_SMELL

Signed-off-by: Peter Suna <peter.suna@pantheon.tech>

### CODE REVIEW

File: src/main/java/io/lighty/yang/validator/YangContextFactory.java [hunk 1]
```diff
@@ -163,7 +163,7 @@ private static String readFile(final String path) throws IOException {
         final List<File> yangFiles = new ArrayList<>();
         File[] files = dir.listFiles();
         if (files == null) {
-            return null;
+            return Collections.EMPTY_LIST;
```
Comment (rovarga) [inline src/main/java/io/lighty/yang/validator/YangContextFactory.java:166]: Collections.emptyList() or List.of() (JDK 11)

File: src/main/java/io/lighty/yang/validator/formats/Analyzer.java [hunk 2]
```diff
@@ -42,12 +42,12 @@ private void printOut() {
         }
     }
 
-    private void analyzeSubstatement(final DeclaredStatement subStatement) {
+    private void analyzeSubstatement(final DeclaredStatement<?> subStatement) {
         String name = subStatement.statementDefinition().getStatementName().getLocalName();
         counter.compute(name, (key, val) -> (val == null) ? 1 : val + 1);
-        final Collection substatements = subStatement.declaredSubstatements();
+        final Collection<?> substatements = subStatement.declaredSubstatements();
```
Comment (rovarga) [inline src/main/java/io/lighty/yang/validator/formats/Analyzer.java:48]: this is not correct: declaredSubstatements() returns a properly-bound collection -- no need to do a ton of these typecasts!

**Review by rovarga — CHANGES_REQUESTED** [review]

File: src/main/java/io/lighty/yang/validator/formats/Analyzer.java (hunk 2 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/Analyzer.java]: Done

File: src/main/java/io/lighty/yang/validator/YangContextFactory.java (hunk 1 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/YangContextFactory.java]: Done

**Review by vainyksi — CHANGES_REQUESTED** [review]
these commits mixes context of changes, they need to be definitely split to separate PRs

please create separate PRs from these commits:
- all the refactors related to method complexity into separate PRs
- other code-smells & checkstyles (at least one PR, but appreciated if will be separated & grouped by types)

**Comment by vainyksi** [comment]
split into #18 & #19

**Comment by vainyksi** [comment]
split into #18 & #19

**Pull request closed without merging**